calc.rb

```diff
@@ -0,0 +1,64 @@
+# define add method
+def add(a, b)
+  return a + b
+end
+
+# define subtract method
+def subtract(a, b)
+  return a - b
+end
+
+# define multiply method
+def multiply(a, b)
+  return a * b
+end
+
+# define divide method
+def divide(a, b)
+  if b != 0
+    return a / b
+  elsif b == 0
+    puts "NOT A NUMBER"
+  end
+end
+
+# define method to check if input is an Integer
+def is_integer(n)
+  n.to_s =~ /[0-9]+/
+  # n.is_a? Integer
+end
+
+puts "What kind of math would you like to do (+, -, *, or /)?"
+operation = gets.chomp
+
+puts "Give me a number:"
+num1 = gets.chomp
+
+# checks for erroneous number input
+while is_integer(num1) == nil
+  puts "NAN"
```

Comment:
This certainly works, but a cleaner way to write it might be `until is_integer(num1)`. No check against `nil` required.